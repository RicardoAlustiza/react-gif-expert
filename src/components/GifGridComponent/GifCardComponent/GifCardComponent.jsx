import PropTypes from 'prop-types';

export const GifCardComponent = ({title, url, id}) => {
  return (
    <div className='card' >
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifCardComponent.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
