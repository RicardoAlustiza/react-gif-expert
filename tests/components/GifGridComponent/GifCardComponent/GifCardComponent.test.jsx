import { render, screen } from '@testing-library/react';
import { GifCardComponent } from '../../../../src/components/GifGridComponent/GifCardComponent/GifCardComponent';

describe('GifCardComponent', () => {

    const title = 'Title';
    const url = 'https://localhost/image.jpg';

    test('should match with the snapshot', () => {

        const { container } = render(<GifCardComponent title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    })

    test('should shown the image with the correct url and alt', () => {

        render(<GifCardComponent title={ title } url={ url } />);

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    })

    test('should shown the title in a paragraph', () => {
            
            render(<GifCardComponent title={ title } url={ url } />);
            expect( screen.getByText(title) ).toBeTruthy();
        }
    )
})