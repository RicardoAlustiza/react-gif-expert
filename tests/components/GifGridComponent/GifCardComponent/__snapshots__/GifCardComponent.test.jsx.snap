// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCardComponent should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="https://localhost/image.jpg"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
